File: bluediamond/normal/NA/102.java
#include <bits/stdc++.h>

using namespace std;

typedef long long ll;
typedef long double ld;

int n;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cin>>n;
    if(n%2==0)
    {
        cout<<n/2<<"\n";
        for(int i=1;i<=n/2;i++)
            cout<<"2 ";
        cout<<"\n";
        return 0;
    }
    cout<<n/2<<"\n";
    for(int i=1;i<n/2;i++)
        cout<<"2 ";
    cout<<"3 \n";
    return 0;
}
/**

**/

File: dormi/normal/NA/102.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long int lli;
bool check(string a){
	int ans=0;
	for(char i:a){
		ans+=(i-'0');
	}
	return !(ans%4);
}
int main(){
    cin.tie(NULL);
    ios_base::sync_with_stdio(false);
    string a;
    cin>>a;
    for(;;){
    	if(check(a)){
    		cout<<a;
    		return 0;
    	}
    	else{
    		a=to_string(stoi(a)+1);
    	}
    }
    return 0;
}